Adding new definition for UAP and UAPAOT builds and adding it to pipebuild

joperezr opened the pull request:
cc: @chcosta @weshaggard 
FYI: @joshfree 

This PR:
 - Sets up a new build definition called DotNet-CoreFx-Trusted-Windows-UWP that is very similar to the one we currently have for Windows except that it doesn't run build-tests nor it triggers jobs for Helix.
 - Change pipebuild.json in order to add these additional configurations to our official build. It adds 16 builds in total (4 build architectures<x86 x64 arm arm64> * 2 verticals<uap and uapaot> * 2 flavors<Debug and Release>)

chcosta commented on buildpipeline/pipeline.json:
```diff
@@ -278,6 +278,136 @@
+            "Type": "build/product/",
+            "Platform": "arm",
+            "ConfigurationGroup": "Release",
+            "SubType": "Uap Vertical"
```
maybe just make this "uap", "uap vertical" seems redundant

chcosta commented:
I wouldn't gate on CI for this, CI doesn't exercise these changes at all

weshaggard commented on buildpipeline/pipeline.json:
```diff
@@ -278,6 +278,136 @@
+        {
+          "Name": "DotNet-CoreFx-Trusted-Windows-UWP",
+          "Parameters": {
+            "PB_Platform": "arm64",
```
We aren't supporting arm64 yet for UWP so we can probably exclude this one for now.

joperezr replied:
is that true for both uap and uapaot? or just uapaot?

weshaggard replied:
both

weshaggard commented on buildpipeline/pipeline.json:
```diff
@@ -278,6 +278,136 @@
+          "Name": "DotNet-CoreFx-Trusted-Windows-UWP",
+          "Parameters": {
+            "PB_Platform": "arm64",
+            "PB_BuildArguments": "-framework=uap -buildArch=arm64 -Release -- /p:SignType=real /p:RuntimeOS=win10",
```
Any reason we aren't parameterizing some of these in the UWP definition as opposed to crafting all them and duplicating them for each leg?

joperezr replied:
well my idea was to keep the definitions as close as possible to the Windows one, since once we are at the point were build-tests.cmd works we basically just want to change the definition back and use the same Windows one. This UWP temp one is just because for now I had to disable two steps of the process. If you still feel it would be better to pass this ones as parameters I can change that.

weshaggard replied:
I see I guess I haven't looked closely at the windows definition. Perhaps this is feedback for @chcosta then as our windows one should be updated to parameterized accordlingly as well.

weshaggard commented on buildpipeline/DotNet-CoreFx-Trusted-Windows-UWP.json:
```diff
@@ -0,0 +1,487 @@
+{
+  "build": [
```
Just for consistency can you call this UAP instead of UWP. I know they are generally considered the same but I prefer the consistency.

joperezr replied:
Sure I'll change it.

weshaggard commented on buildpipeline/DotNet-CoreFx-Trusted-Windows-UWP.json:
```diff
@@ -0,0 +1,487 @@
+      },
+      "inputs": {
+        "filename": "$(Build.SourcesDirectory)\\corefx\\build.cmd",
+        "arguments": "$(PB_BuildArguments)",
```
I would prefer this be `-framework=uap -buildArch=arm64 -Release -- /p:SignType=real /p:RuntimeOS=win10` with the framework and arch parameterized, and the signType should be parameterized so that we could kick off a build that isn't real signed.

weshaggard commented on buildpipeline/pipeline.json:
```diff
@@ -278,6 +278,136 @@
+            "PB_SyncArguments": "-p -- /p:ArchGroup=x64 /p:RuntimeOS=win10",
+            "PB_CreateHelixArguments": "/t:CloudBuild /p:ArchGroup=x64 /p:ConfigurationGroup=Release /p:\"EnableCloudTest=true\" /p:\"BuildMoniker=none\" /p:\"TargetQueue=Windows.10.Amd64\" /p:\"TestProduct=corefx\" /p:\"TimeoutInSeconds=1200\" /p:\"TargetOS=Windows_NT\" /p:FilterToOSGroup=Windows_NT"
+          },
+          "ReportingParameters": {
```
@chcosta what is the difference between parameters and reporting parameters?

joperezr replied:
AFAIK ReportingParameters are the ones used by mission control to correctly name them and display them.

chcosta replied:
What Joe said

weshaggard commented on buildpipeline/pipeline.json:
```diff
@@ -278,6 +278,136 @@
+          "Parameters": {
+            "PB_Platform": "x86",
+            "PB_TargetQueue": "Windows.10.Amd64",
+            "PB_BuildArguments": "-framework=uapaot -buildArch=x86 -Release -- /p:SignType=real /p:RuntimeOS=win10",
```
The RuntimeOS should match the RID, which I don't believe are all win10 based

joperezr replied:
for this particular one I just copied whatever we were using for the Windows definition. @chcosta any ideas why would x86 pass in RuntimeOS?

weshaggard replied:
I believe @chcosta passed in RuntimeOS because the build machines were running on win81 and we wanted the RID to be win10 but we need to verify if that is going to cause our RID specific packaging to be broken.

chcosta replied:
Wes is correct about the reasoning

joperezr commented:
I've addressed most of the feedback except for that piece where the builds should be parametarized instead,

joperezr commented:
I've addressed most of the feedback except adding parameters to the build definitions since the idea for now was to keep it as close as possible to the Windows one. I'll merge this now to start producing the package and will address any more feedback later.
